pkg-manager/plugin-commands-installation/test/catalog.ts

```diff
@@ -0,0 +1,406 @@
+import { getConfig } from '@pnpm/config'
+import { WANTED_LOCKFILE } from '@pnpm/constants'
+import { createPeersDirSuffix } from '@pnpm/dependency-path'
+import { type Lockfile } from '@pnpm/lockfile-types'
+import { type ProjectId, type ProjectManifest } from '@pnpm/types'
+import { filterPackagesFromDir } from '@pnpm/workspace.filter-packages-from-dir'
+import { preparePackages } from '@pnpm/prepare'
+import { install } from '@pnpm/plugin-commands-installation'
+import path from 'path'
+import readYamlFile from 'read-yaml-file'
+import writeYamlFile from 'write-yaml-file'
+import { DEFAULT_OPTS } from './utils'
+
+/**
+ * A utility to make writing catalog tests easier by reducing boilerplate.
+ */
+class CatalogTestsController {
+  private readonly workspaceDir: string
+
+  constructor (readonly pkgs: Array<{ location: string, package: ProjectManifest }>) {
+    preparePackages(pkgs)
+    this.workspaceDir = process.cwd()
+  }
+
+  async writeWorkspaceYml (data: unknown) {
+    await writeYamlFile(path.join(this.workspaceDir, 'pnpm-workspace.yaml'), data)
```

Comment:
use sync functions in tests. But this won't be needed in `core`, where you will pass catalogs through options.

Reply:
Thanks.

Reply:
Done! Latest push moves these tests to `core`.